File: CoreJava/SOLID-princples-app/src/isp/violation/Robot.java
package isp.violation;

public class Robot implements IWorker {

	@Override
	public void startsWork() {
		System.out.println("Robot starts working");
	}

	@Override
	public void endsWork() {
		System.out.println("Robot ends working");
	}

	@Override
	public void startsEat() {
		throw new RuntimeException("Robots do not eat");
	}

	@Override
	public void endsEat() {
		throw new RuntimeException("Robots do not eat");
	}

}

File: DotNET/Solid Principles/ISPViolationApp/ISPViolationApp/Robot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ISPViolationApp
{
    class Robot : IWorkable
    {
        public void EndsEat()
        {
            Console.WriteLine("Robot ends eating");
        }

        public void EndsWork()
        {
            Console.WriteLine("Robot ends working");
        }

        public void StartsEat()
        {
            Console.WriteLine("Robot starts eating");
        }

        public void StartsWork()
        {
            Console.WriteLine("Robot starts working");
        }
    }
}
